cornell-dti/course-plan
# Fix overridden fulfillments race condition in course conflict modal

### Summary 






This pull request fixes the overridden fulfillments race condition seen in #692. In addition, it
- [x] renames `override-fulfillment-choices.ts` to `overridden-fulfillment-choices.ts`, as mentioned in #689
- [x] removes the comparison to `originalReqsPerConflict` (we shouldn't need to compare to the original state in order to update the user data)

#### TODO  

We might want to change how we represent `selectedReqsPerConflict` in `CourseConflictModal.vue`. For example, instead of mapping reqName => boolean, we could store two sets (one for selected requirements and one for unselected requirements).

We need to add some e2e tests for this modal.




### Test Plan 


1. Enable requirements conflicts in the browser console, i.e. `GK.enableRequirementConflicts()`
2. Add CS 4152 as a CS major (or any other class that would cause a conflict
3. Open the conflict modal by clicking on "FIX NOW" in the warnings
4. Select different combinations of requirements and make sure they save properly and are reflected in the requirements graph/debugger (`GK.enableRequirementDebugger()`).
